molgenis-data-annotators/src/main/java/org/molgenis/data/support/AnnotatorDependencyOrderResolver.java

```diff
@@ -24,7 +25,12 @@
 		Queue<RepositoryAnnotator> sortedList = new LinkedList<>();
 		for (RepositoryAnnotator annotator : requestedAnnotatorList)
 		{
-			if (!sortedList.contains(annotator))
+			if (annotator instanceof EffectsAnnotator)
+			{
+				// FIXME: implement correct dependency resolving for Effect annotator
```

Comment:
Potentially very complex, not needed now. Workaround works fine for current situation but it will be a nice enhancement for the future.